lib/redirect_map.rb

```diff
@@ -0,0 +1,23 @@
+require 'yaml'
+require 'singleton'
+
+class RedirectMap
+  include Singleton
+
+  attr_reader :redirects
+
+  def initialize
+    @redirects ||= {}
+  end
+
+  def load(filename)
+    @redirects = YAML.load(File.read(filename))
+    fail InvalidRedirectMapFile, "redirect map file \"#{filename}\" must be a YAML file of key-value pairs" unless @redirects.respond_to?(:key?)
+  end
+
+  def lookup(url)
+    @redirects[url]
+  end
```

Comment:
i'm actually not a fan of that idiom. To me, when I see `some_obj["foo"]` I think that `some_obj` is an `Enumerable`. It's confusing to me if you can do `some_obj["foo"]`, but you get an error when you do `some_obj.each`, or `som_obj.size`.